### DISCUSSION TITLE
Richer logging for parallel SMC Sampler

### DISCUSSION THREAD

**Discussion by jucor** [pymc-devs/pymc3#7080] [Ideas]
The current SMC sampler only logs progress (and beta schedule) of the first worker. This is possibly because the underlying `fastprogress` does not support multiple progress bars in parallel, see https://github.com/fastai/fastprogress/issues/80 and https://github.com/fastai/fastprogress/issues/39.

This is not an issue is all processes work roughly at the same speed. However that might not always be the case, especially if one of the processes are sampling in a region of parameter space that cause slow likelihood evaluation.  The first process, which is logging, might then reach 100% long before the last process. Real example: ODEs with varying stiffness depending on the parameter values. Ask me how I know 😆

It would therefore be handy to:
- have one progress bar for each "chain"/process. That would require switching to another progressbar library.
- or log more information to the logger, e.g. in debug mode. The logger is already accessible and works well accross processes. I could open a PR for that, actually :) This solution, besides being less work, would be in line with https://github.com/pymc-devs/pymc/issues/6691 .

**Comment by jucor** [comment]
(I am somewhat familiar with the SMC sampler's progress/logging now that I've completed #7079 )

**Comment by zaxtax** [comment]
@jucor Does the following PR address your logging concerns? https://github.com/pymc-devs/pymc/pull/7233

**Comment by jucor** [comment]
Woops, apologies, only seeing this (didn't receive the notification, sorry).
As per https://github.com/pymc-devs/pymc/pull/7410 , it somewhat does, thank you! :) Closing this !
